app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx

```diff
@@ -180,6 +217,23 @@ export class CreatePipelineModal extends React.Component<ICreatePipelineModalPro
     return () => this.setState({useTemplate});
   }
 
+  private handleUseManagedTemplateSelection(useManagedTemplate: boolean): () => void {
+    return () => {
+      this.setState({
+        useManagedTemplate,
+        templateSourceUrl: '',
+        loadingTemplateFromSourceError: false,
+        command: Object.assign({}, this.state.command, {template: null}),
+      });
+    };
+  }
+
+  public handleSourceUrlChange(e: React.ChangeEvent<HTMLInputElement>): void {
+    const templateSourceUrl = e.target.value;
+    this.setState({templateSourceUrl});
+    this.loadPipelineTemplateFromSource(templateSourceUrl);
```

Comment:
Sorry, poor clarity (and also and a poor understanding of the default options debounce) on my part, I was using debounced as a verb at runtime. I meant that if `wait` had expired for the debounced function.  I thought `leading` was the default behavior, not `trailing`. I have much less concern for back-to-back `setState` calls with a `trailing` debounce. So nothing to see here.